### ISSUE TITLE
Dump to stout

### ISSUE THREAD

**Issue by inode64** [abbbi/virtnbdbackup#16]
It is possible to make a full backup and that this is sent stdout instead of a directory
To be able to send it for example to borgbackup

Thanks for you good job!!

**Comment by abbbi** [comment]
hi, 

yes, but it does only make sense if used with option -i, so only the backup for a single disk is started.
If it processes all disks, the stream for each disk is concat'ed after each other and during restore it will be troublesome
to differenciate between disks.

Example command would be:

` virtnbdbackup -d vm1 -i vda -o - > <targetfile>`

It then stores the backup log and all required checkpoint information in the directory it is executed in.
In theory it would also work with incremental backups, but then you need a seperate directory for each disk that is kept between runs.

**Comment by abbbi** [comment]
> yes, but it does only make sense if used with option -i, so only the backup for a single disk is started.

and it does not work with raw disks because backup of raw disks means it saves the data "as is" and needs
to seek to the same positions, which is of course not working if it writes to stdout or fifo pipes.

**Comment by inode64** [comment]
Thanks, I carry out some tests and I tell you

**Comment by abbbi** [comment]
> Thanks, I carry out some tests and I tell you

i fixed some things in the master branch regarding the include option, and it now also saves the virtual machine configuratio to the folder the command is started from. The configuration is required vor the restore utility to work because it parses the vm disk information from the saved xml file.

It might be worth a try to implement some output stream class that for example would enable things like:

`virtnbdbackup -d vm1 -l full -o foo.tar`

-> would then create a tar file with all the contents (backup log, checkpoint information, vm config, data of the disks)

**Comment by inode64** [comment]
An idea, it would also be to be able to make the backup of the tar in stdout

As an example

`virtnbdbackup -d vm1 -l full --archive tar -o -`

or

`virtnbdbackup -d vm1 -l full --archive tar -o /srv/vm/vm1.tar `
`virtnbdbackup -d vm1 -l full --compress lz4 --archive tar -o /srv/vm/vm1.tar.lz4`
`virtnbdbackup -d vm1 -l full  --archive zstd -o /srv/vm/vm1.tar.zstd `


Then you can include different types of files, tar, zip, raw, ...

**Comment by abbbi** [comment]
yes, that would be neat. Pythons tarfile also supports compression. Maybe i have some spare time next week to look at something like this. It should be implemented in a way that there is maybe a TargetFolderClass and a TargetTarClass that returns the required write(), read() and maybe seek() (not sure about tarfile) functions for both utilites to use them in a transparent way. Tarfile needs to be sure to use the streaming mode to be sure not to exceed memory usage during backup/restore.

**Comment by inode64** [comment]
Thank you, I will be waiting for the improvements

**Comment by abbbi** [comment]
Creating a tarfile in streaming mode with the current concept is a little bit more complicated than expected. The Problem is that the tarfile module in streaming mode expects a fixed size for the tar file before one can actually start writing the data. Currently see no good way to have the size of the resulting backup file in streaming format pre-calculated. Another solution would be to cache the data in memory, which results in high memory usage during backup :(
